Mathlib/CategoryTheory/Adjunction/Mates.lean

```diff
@@ -127,6 +175,240 @@ theorem unit_transferNatTrans (f : G ⋙ L₂ ⟶ L₁ ⋙ H) (X : C) :
 -- See library note [dsimp, simp]
 end Square
 
+-- ER: A new section proving that the mates bijection commutes with vertical composition of squares.
+section MatesVComp
+
+variable {A : Type u₁} {B : Type u₂} {C : Type u₃}
+variable {D : Type u₄} {E : Type u₅} {F : Type u₆}
+variable [Category.{v₁} A] [Category.{v₂} B][Category.{v₃} C]
+variable [Category.{v₄} D] [Category.{v₅} E][Category.{v₆} F]
+variable {G₁ : A ⥤ C}{G₂ : C ⥤ E}{H₁ : B ⥤ D}{H₂ : D ⥤ F}
+variable {L₁ : A ⥤ B}{R₁ : B ⥤ A} {L₂ : C ⥤ D}{R₂ : D ⥤ C}
+variable {L₃ : E ⥤ F}{R₃ : F ⥤ E}
+variable (adj₁ : L₁ ⊣ R₁) (adj₂ : L₂ ⊣ R₂) (adj₃ : L₃ ⊣ R₃)
+
+def LeftAdjointSquare.vcomp :
+    (G₁ ⋙ L₂ ⟶ L₁ ⋙ H₁) → (G₂ ⋙ L₃ ⟶ L₂ ⋙ H₂) →
+    ((G₁ ⋙ G₂) ⋙ L₃ ⟶ L₁ ⋙ (H₁ ⋙ H₂)) := fun α β ↦
+  (whiskerLeft G₁ β) ≫ (whiskerRight α H₂)
+
+def RightAdjointSquare.vcomp :
+    (R₁ ⋙ G₁ ⟶ H₁ ⋙ R₂) → (R₂ ⋙ G₂ ⟶ H₂ ⋙ R₃) →
+    (R₁ ⋙ (G₁ ⋙ G₂) ⟶ (H₁ ⋙ H₂) ⋙ R₃) := fun α β ↦
+  (whiskerRight α G₂) ≫ (whiskerLeft H₁ β)
+
+theorem Mates_vcomp
+  (α : G₁ ⋙ L₂ ⟶ L₁ ⋙ H₁)
+  (β : G₂ ⋙ L₃ ⟶ L₂ ⋙ H₂) :
+  (Mates (G := G₁ ⋙ G₂) (H := H₁ ⋙ H₂) adj₁ adj₃) (LeftAdjointSquare.vcomp α β)
```

Comment:
Does Lean need G and H to be spelled out here to figure out what's going on? If so, maybe they should be explicit arguments in `Mates`

Reply:
It seems like it needs it only sometimes, so I like keeping these implicit.